guidelines/github-contribution-guide.md

```diff
@@ -0,0 +1,266 @@
+
+```
+git checkout -b <feature_branch_name>
+```
+
+Now that you have created a local feature branch, you can perform your updates.
+
+## Committing and Pushing Changes to Your Forked Repository
+
+Make the desired changes to files in your feature branch. You can now commit this code and push it to your forked repository:
+
+- Add existing files you have changed to your commit by executing the following `git add` command
+- The `-p` flag will open an interactive terminal for you to review and approve your changes before staging them for your commit.
+- Alternatively, use the `-u` flag to stage all updated files for your commit.
+
+```
+git add -p
+```
+
+- Add any new files you have created by executing:
+
+```
+git add <file1> <file2>
+```
+
+- You can now create your commit containing the changes you just added.
+- Your commit message should contain the following information:
+  - A one line commit title summarizing the commit, followed by an empty line
+  - A multi-line commit message body explaining the need for the change and how you approached it
+  - A link to GitHub issue (if exists), using syntax like `Resolves #<GitHub issue number>` so that the GitHub issue gets linked to the pull request.
+  - The `-s` flag will automatically sign the commit
+
+```
+git commit -s
+```
+
+- You can now push your local changes to your forked repository
+
+```
+git push origin <feature_branch_name>
+```
+
+Note - If you want to integrate upstream changes from the original repository
+before pushing your changes see [Syncing Your Fork With the Upstream
+Repository](#syncing-your-fork-with-the-upstream-repository).
+
+You have now successfully pushed your local changes to your forked repository.
+To integrate these changes you must now go through the pull request process.
+
+## Opening a Pull Request in GitHub
+
+Now that you've created and pushed changes to a feature branch in your forked repository,
+you can now open a pull request against the original Hyperledger repository from which you created your fork and begin the code review process.
+
+- Navigate to https://github.com/hyperledger/<original_repository>
+- Select the Pull Request tab at the top of the page
+- In the top right corner of the Pull Requests page, select New Pull Request
+- On the Compare Changes page, select "compare across forks" at the top of the page
+- Select the Hyperledger repository and branch that you want to merge into as the "base repository" and "base" branch
+- Select your fork and your feature branch as the "head repository" and "compare" branch
+- Select Create Pull Request
+- You'll have the option to open your pull request as a Draft if you are not yet ready to have it reviewed.
+
+Congratulations, you have now submitted your first pull request to a Hyperledger project.
+Your pull request will now run through CI checks.
+You can monitor your pull request CI progress by navigating to the Checks tab of the pull request.
+
```

Comment:
We should address the case where the CI fails and then I think it'd be worth reinforcing the need to pay attention to the feedback received from reviewers and respond. Too often people just submit their PR and disappear...
```suggestion
If the CI fails because you introduced some error, you need to update your pull request to address the problem. Maintainers are unlikely to pay attention to a pull request for which the CI fails.

If the CI passes, be on the lookout for comments from the reviewers and be responsive. It is the project's maintainers responsibility
to review your pull request but it is yours to respond to their comments. In particular reviewers may request that you make some
changes to your commit. Not acting on these requests will most likely leave your pull request in limbo.

```

Reply:
Resolved!